test/ruby/test_hash.rb

```diff
@@ -304,6 +304,17 @@ def test_AREF_fstring_key
     assert_equal before, ObjectSpace.count_objects[:T_STRING]
   end
 
+  def test_AREF_fstring_key_default_proc
+    h = Hash.new do |h, k|
+      k.frozen?
+    end
+
+    str = "foo"
+    refute str.frozen? # assumes this file is frozen_string_literal: false
```

Comment:
```suggestion
    str = +"foo"
```

Reply:
I think it's important to check, because if we're running under `frozen_string_literal: true`, we won't be testing opt_aref_with (which only exists under frozen_string_literal: false). Alternatively I could put this in an `assert_separately`